File: Clase12/ProyectosNET/SoftProgSolution/SoftProgPersistencia/DAO/IDAO.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SoftProgPersistencia.DAO
{
    public interface IDAO <T>
    {
        int insertar(T objeto);
        int modificar(T objeto);
        int eliminar(int id);
        T buscarPorId(int id);
        List<T> listarTodos();
    }
}

File: Lab9Calificado/Lab11_SOAP_P3_RENIEC/ReniecSoft/ReniecSoftPersistance/src/main/java/pe/edu/pucp/reniecsoft/persistance/dao/IDAO.java
package pe.edu.pucp.reniecsoft.persistance.dao;

import java.util.List;

public interface IDAO <T>{
      int insertar(T objeto);
      int modificar(T objeto);
      int eliminar(String dni);
      T buscarPorDni(String dni);
      List<T> listarTodos();
}
